django/db/models/sql/query.py

```diff
@@ -2090,7 +2090,14 @@ def add_deferred_loading(self, field_names):
             self.deferred_loading = existing.union(field_names), True
         else:
             # Remove names from the set of any existing "immediate load" names.
-            self.deferred_loading = existing.difference(field_names), False
+            new_existing = existing.difference(field_names)
+            if not new_existing:
```

Comment:
If you handle the positive case first, this can be the simpler `if new_existing:`. You could also use an assignment expression to simplify further (`if new_existing := existing.difference(field_names):`).

Reply:
Agreed, you can use the walrus operator even twice:
```python
            if new_existing := existing.difference(field_names):
                self.deferred_loading = new_existing, False
            else:
                self.clear_deferred_loading()
                if new_only := set(field_names).difference(existing):
                    self.deferred_loading = new_only, True
```